### ISSUE TITLE
sfcwater_fracliq in ED2

### ISSUE THREAD

**Issue by Viskari** [EDmodel/ED2#140]
I am at the moment struggling with an issue and I thought to ask if anyone here had insight what might be causing it.

When I run the current version of ED2, the sfcwater_fracliq is 0 in all layers despite being in a location with snow. As our interest is with the albedo effects, this is causing us issues. Any thoughts on the matter would be appreciated.

**Comment by rgknox** [comment]
Hi @Viskari 

fracliq is the fraction of the surface water in each layer that is liquid compared to a whole constituted of liquid and solid.  If the location is cold enough, then the liquid fraction should reach 0 as the water temperature drops below its freezing point.  

Perhaps you interpreted this variable as the area fraction that is covered by water?  Am I understanding your question correctly?  Or is your argument more that there should be a minuscule amount of liquid water bound within the snow?

**Comment by Viskari** [comment]
No, what I meant is that in radiate_driver currently uses sfcwater_fracliq
to calculate the inclusion of snow in the surface albedo. The way you
describe fracliq makes sense, but then I again don't understand why it used
for the snow calculations in radiate_driver. The exact line is:

         albedo_sfcw_par = albedo_damp_par +
csite%sfcwater_fracliq(ksn,ipa)               &
                                           * ( snow_albedo_vis -
albedo_damp_par )

And even the description says that this is the place where snow albedo is
at the moment included.

**Comment by mdietze** [comment]
Looks like that's just a weighted mean between the 'damp' albedo and the 'snow' albedo, but that whomever implemented it got the order reversed (i.e. when it's snow it uses the damp, and when it's liquid it uses the snow). I'd just reverse all the damp and snow terms.

**Comment by mpaiao** [comment]
@Viskari I agree with you, the code is wrong, it should be:
```fortran
albedo_sfcw_par = snow_albedo_vis + csite%sfcwater_fracliq(ksn,ipa) * (albedo_damp_par - snow_albedo_vis)
albedo_sfcw_nir = snow_albedo_nir + csite%sfcwater_fracliq(ksn,ipa) * (albedo_damp_nir - snow_albedo_nir)
```
Same for NIR.  Emissivity looks right, but it's worth checking.  Apologies for the confusion.

**Comment by Viskari** [comment]
No problem, just glad to have caught the problem.

Marcos, do I fix it or do you want to correct the code?

**Comment by mpaiao** [comment]
Would you mind submitting the pull request? My code has some other changes I'm testing right now.

**Comment by Viskari** [comment]
Sure, no problem. I'll have it up later today.

**Comment by xiangtao-princeton** [comment]
It seems the bug is fixed in the ED code but still exists in **EDR** and **BRAMS**. Would this be of concern for the users of those two? @Viskari @mpaiao
